src/IonEventStream.ts

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+import { Reader } from "./IonReader";
+import {IonContainerEvent, IonEndEvent, IonEvent, IonEventFactory} from "./IonEvent";
+import { Writer } from "./IonWriter";
+import { IonType } from "./IonType";
+import { IonTypes } from "./IonTypes";
+import { IonEventType } from "./IonEvent";
+import {TypeCodes} from "./IonBinary";
+import {TextReader} from "./IonTextReader";
+import {makeReader} from "./Ion";
+import { Span } from "./IonSpan";
+import {BinaryReader} from "./IonBinaryReader";
+
+export class IonEventStream {
+    private eventStream : IonEvent[];
+    private reader : Reader;
+    private eventFactory : IonEventFactory;
+
+    constructor(reader : Reader) {
+        this.eventStream = [];
+        this.reader = reader;
+        this.eventFactory = new IonEventFactory();
+        this.generateStream();
+
+    }
+
+    writeEventStream(writer : Writer) {
+        writer.writeSymbol("ion_event_stream");
+        for(let i : number = 0; i < this.eventStream.length; i++) {
+            this.eventStream[i].write(writer);
+        }
+    }
+
+    writeIon(writer : Writer) {
+        let tempBuf : number[];
+        let tempEvent : IonEvent;
+        for(let indice : number = 0; indice < this.eventStream.length; indice++){
```

Comment:
indice is not plural

Reply:
Indice is equivalent to index in most dictionaries (english sucks). My first CS prof delineated between the position and the value at said position since index/indice ambiguously mean both.

Reply:
I think its a math thing

Reply:
This keeps distracting people. Pragmatically, it might be advisable to switch to "index" if for no other reason than to not have to explain it during every review.